types/seat/wlr_seat_keyboard.c

```diff
@@ -409,8 +414,28 @@ void seat_client_create_keyboard(struct wlr_seat_client *seat_client,
 	seat_client_send_keymap(seat_client, keyboard);
 	seat_client_send_repeat_info(seat_client, keyboard);
 
-	// TODO possibly handle the case where this keyboard needs an enter
-	// right away
+	struct wlr_seat_client *focused_client =
+		seat_client->seat->keyboard_state.focused_client;
+	struct wlr_surface *focused_surface =
+		seat_client->seat->keyboard_state.focused_surface;
+
+	// Send an enter event if there is a focused client/surface stored
+	if (focused_client != NULL && focused_surface != NULL) {
+		struct wl_array keys;
+		wl_array_init(&keys);
```

Comment:
The keys array should be populated from the keyboard state.

Reply:
Whoops, should be fixed by #2068